pallets/extrinsic-filter/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+		/// Set the mode
+		///
+		/// The storage of `BlockedExtrinsics` is unaffected.
+		/// The reason is we'd rather have this pallet behave conservatively:
+		/// having extra blocked extrinsics is better than having unexpected whitelisted extrinsics.
+		/// See the test `set_mode_should_not_clear_blocked_extrinsics()`
+		///
+		/// Weights should be 2 DB writes: 1 for mode and 1 for event
+		#[pallet::weight(2 * T::DbWeight::get().write)]
+		pub fn set_mode(origin: OriginFor<T>, mode: OperationalMode) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+
+			// we don't care which mode we are currently in,
+			// just set to the desired mode
+			Mode::<T>::put(mode);
+
+			Self::deposit_event(Event::ModeSet { new_mode: mode });
+
+			Ok(().into())
+		}
+
+		/// block the given extrinsics
+		/// (pallet_name_bytes, function_name_bytes) can uniquely identify an extrinsic
+		/// if function_name_bytes is None, all extrinsics in `pallet_name_bytes` will be blocked
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn block_extrinsics(
+			origin: OriginFor<T>,
+			pallet_name_bytes: Vec<u8>,
+			function_name_bytes: Option<Vec<u8>>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			let pallet_name_string = sp_std::str::from_utf8(&pallet_name_bytes)
+				.map_err(|_| Error::<T>::CannotConvertToString)?;
+			// we disallow blocking this pallet itself
+			ensure!(
+				pallet_name_string != <Self as PalletInfoAccess>::name(),
+				Error::<T>::CannotBlock
+			);
+
+			BlockedExtrinsics::<T>::mutate(
+				(pallet_name_bytes.clone(), function_name_bytes.clone().unwrap_or_default()),
+				|v| {
+					if v.is_none() {
+						*v = Some(());
+						Self::deposit_event(Event::ExtrinsicsBlocked {
+							pallet_name_bytes,
+							function_name_bytes,
+						});
+						// do not pay the fee upon successful block
+						Ok(Pays::No.into())
+					} else {
+						Err(Error::<T>::ExtrinsicAlreadyBlocked.into())
+					}
+				},
+			)
+		}
+
+		/// unblock the given extrinsics
+		/// (pallet_name_bytes, function_name_bytes) can uniquely identify an extrinsic
+		/// if function_name_bytes is None, all extrinsics in `pallet_name_bytes` will be unblocked
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn unblock_extrinsics(
+			origin: OriginFor<T>,
+			pallet_name_bytes: Vec<u8>,
+			function_name_bytes: Option<Vec<u8>>,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			if BlockedExtrinsics::<T>::take((
+				pallet_name_bytes.clone(),
+				function_name_bytes.clone().unwrap_or_default(),
+			))
+			.is_some()
+			{
+				Self::deposit_event(Event::ExtrinsicsUnblocked {
+					pallet_name_bytes,
+					function_name_bytes,
+				});
+			} else {
+				return Err(Error::<T>::ExtrinsicNotBlocked.into())
+			}
+			// do not pay the fee upon successful unblock
+			Ok(Pays::No.into())
+		}
+	}
+
+	impl<T: Config> Contains<T::Call> for Pallet<T>
```

Comment:
This Contains trait is implemented for the ExtrisicFilter pallet, but where is this one used?

Reply:
> This Contains trait is implemented for the ExtrisicFilter pallet, but where is this one used?

In runtime binding